src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs

```diff
@@ -0,0 +1,995 @@
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        // PROTOTYPE(ft): is it fine that records parse as a single declaration here, but not other type kinds?
+        [Theory]
+        [InlineData(SyntaxKind.RecordKeyword)]
+        public void FileModifier_04(SyntaxKind typeKeyword)
+        {
+            UsingNode($$"""
+                partial file {{SyntaxFacts.GetText(typeKeyword)}} C { }
+                """);
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxFacts.GetBaseTypeDeclarationKind(typeKeyword));
+                {
+                    N(SyntaxKind.PartialKeyword);
+                    N(SyntaxKind.FileKeyword);
+                    N(typeKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void FileModifier_05()
+        {
+            UsingNode($$"""
+                file partial record struct C { }
+                """);
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.RecordStructDeclaration);
+                {
+                    N(SyntaxKind.FileKeyword);
+                    N(SyntaxKind.PartialKeyword);
+                    N(SyntaxKind.RecordKeyword);
+                    N(SyntaxKind.StructKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        // PROTOTYPE(ft): is it fine that records parse here, but not other type kinds?
+        [Fact]
+        public void FileModifier_06()
+        {
+            UsingNode($$"""
+                partial file record struct C { }
+                """);
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.RecordStructDeclaration);
+                {
+                    N(SyntaxKind.PartialKeyword);
+                    N(SyntaxKind.FileKeyword);
+                    N(SyntaxKind.RecordKeyword);
+                    N(SyntaxKind.StructKeyword);
+                    N(SyntaxKind.IdentifierToken, "C");
+                    N(SyntaxKind.OpenBraceToken);
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact]
+        public void FileModifier_07()
+        {
+            UsingNode($$"""
+                file partial ref struct C { }
+                """);
+            N(SyntaxKind.CompilationUnit);
+            {
+                N(SyntaxKind.IncompleteMember);
```

Comment:
>IncompleteMember [](http://example.com/codeflow?start=29&length=16)

Is this expected?

Reply:
It seems perfectly valid to have a `file partial ref struct`. I'll have to fix it to ensure we get a correct parse here.

Reply:
I was wrong. `partial ref struct` is [not allowed](https://sharplab.io/#v2:EYLgtghglgdgNAFxBAzguATEBqAPgJwFMAzAAgAcJ8EoIAbUtfAVwGMFSBlARlIG9SAXwCwAKDGVqtBkTJM2HTgCZ+QoA===). You have to say `ref partial struct`.